variables.tf

```diff
@@ -636,3 +636,51 @@ variable "horizon-plugins" {
   type        = list(string)
   default     = []
 }
+
+variable "enable-validation" {
+  description = "Enable Tempest deployment"
+  type        = bool
+  default     = false
+}
+
+variable "tempest-channel" {
+  description = "Operator channel for Tempest deployment"
+  type        = string
+  default     = null
+}
+
+variable "tempest-revision" {
+  description = "Operator channel revision for Tempest deployment"
+  type        = number
+  default     = null
+}
+
+variable "tempest-config" {
+  description = "Operator config for Tempest deployment"
+  type        = map(string)
+  default     = {}
+}
+
+variable "grafana-agent-k8s-name" {
+  description = "Operator name for Grafana Agent K8s deployment"
+  type        = string
+  default     = "grafana-agent-k8s"
+}
+
+variable "metrics-endpoint" {
```

Comment:
These are some variable definitions (grafana-agent-k8s-name, metrics-endpoint, logging-provider, grafana-dashboards-consumer) to be used by any application in the openstack model. For example, the openstack-exporter (which is currently directly connected to COS model, and we might need to fix it in the later EPIC), and the tempest-k8s. 

The example usage is:
1. when enabling the observability plugin, we fill the [variables ](https://github.com/agileshaw/sunbeam-terraform/pull/1/files/5c615cb9b146fc3271cefed06a10fb2f6ef1bed1#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR900-R928) with appropriate values
2. these values will be updated in the actual terraform plan for the openstack model (i.e. the main.tf)
3. observability plugin will apply the plan with the updated value to build the relation between grafana-agent-k8s and tempest-k8s (and in the later EPIC, also the openstack-exporter-k8s)

Reply:
I know they are not related to COS directly, but I dot understand why you need them instead of using name define in proper .tf file? Why this needs to be configurable?
Not an expert, just to clarify thinks.
BTW, your link is not working again.

Reply:
They need to be added / removed dynamically. You cannot hardcode them

Reply:
Again reason is that we have two repos for final Terraform plan, which is strange, but now I understand why we needed it.